vmdb/app/models/host.rb

```diff
@@ -1393,16 +1393,24 @@ def self.get_hostname(ipAddress)
 
   def connect_ssh(options={})
     require 'MiqSshUtil'
-    if self.has_authentication_type?(:remote)
-      rl_user, rl_password = self.auth_user_pwd(:remote)
-      su_user, su_password = self.auth_user_pwd(:root)
-    else
-      rl_user, rl_password = self.auth_user_pwd(:root)
+    if self.kind_of?(HostOpenstackInfra)
+      # HostOpenstackInfra is using auth key set on ext_management_system level, not individual hosts
+      rl_user, auth_key = self.auth_user_keypair(:keypair)
```

Comment:
@blomquisg @Fryguy POC of using private key, stored on Provider to level, to do ssh fleecing of the hosts. (we will have one key for all host).

What do you think? It was all kinda hardcoded for password. I think I am doing too much conditions everywhere, probably the base auth function should be somehow made more general..